src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/UnicodeRanges.cs

```diff
@@ -32,9 +32,8 @@ private static UnicodeRange CreateEmptyRange(ref UnicodeRange range)
         {
             // If the range hasn't been created, create it now.
             // It's ok if two threads race and one overwrites the other's 'range' value.
-            var newRange = new UnicodeRange(0, 0);
-            Volatile.Write(ref range, newRange);
-            return newRange;
+            var range = new UnicodeRange(0, 0);
+            return range;
```

Comment:
> What else could cause racing threads to see partially instantiated object? 

In theory the compilers/hardware are free to reorder the memory accesses such that the writes to initialize the instance are done after the instance is published; this wouldn't be visible to the initializing thread but could be to others. This is why some forms of double-checked locking are theoretically problematic and why the CLR sometimes incurs extra fences/costs to prevent dev bugs from manifesting. Igor wrote some nice articles on this stuff: https://msdn.microsoft.com/en-us/magazine/jj863136.aspx
Now, this may very well be one of the cases where the CLR will insert extra fences, I'm simple saying you should confirm before you remove the fences. @ericeil will know...